### PR TITLE
Fix ensure by provider id

### LINKED ISSUE #63226: [Cloud controller manager] Node got deleted before providerId is set
**Is this a BUG REPORT or FEATURE REQUEST?**:
/kind bug

**What happened**:
The change here removes exiternal id, but the call to instance id was not made used at all, it will always return false.
https://github.com/kubernetes/kubernetes/pull/60692/files#diff-a1ff61328540d76c791d919f71da4ad5R423

```
// MonitorNode here, calls ensureNodeExistsByProviderIDOrInstanceID, but provider id is not populated
I0427 02:48:54.917288       1 node_controller.go:260] Deleting node since it is no longer present in cloud provider: k8s-master-40525011-0
I0427 02:48:54.917319       1 node_controller.go:268] Recording DeletingNode event message for node k8s-master-40525011-0
I0427 02:48:54.917644       1 event.go:218] Event(v1.ObjectReference{Kind:"Node", Namespace:"", Name:"k8s-master-40525011-0", UID:"7b9fc07c-49c5-11e8-abb4-000d3a179d0f", APIVersion:"", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'Deleting Node k8s-master-40525011-0 because it's not present according to cloud provider' Node k8s-master-40525011-0 event: DeletingNode

// AddCloudNode here, provider id is populated
I0427 02:48:55.147050       1 node_controller.go:375] Adding node label from cloud provider: failure-domain.beta.kubernetes.io/zone=0
I0427 02:48:55.147127       1 node_controller.go:379] Adding node label from cloud provider: failure-domain.beta.kubernetes.io/region=ea

// Trying to set provider id, but the node was just removed.
E0427 02:48:55.240801       1 node_controller.go:396] nodes "k8s-master-40525011-0" not found
```

At this time no node was able to register

**What you expected to happen**:
ProviderId was used in check `ensureNodeExistsByProviderIDOrInstanceID`

**How to reproduce it (as minimally and precisely as possible)**:

**Anything else we need to know?**:

**Environment**:
```
Client Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0-alpha.1", GitCommit:"4761788b2afa42a4573a6794902eb93fe666d5c5", GitTreeState:"clean", BuildDate:"2018-04-19T19:25:45Z", GoVersion:"go1.10.1", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"11+", GitVersion:"v1.11.0-alpha.1", GitCommit:"4761788b2afa42a4573a6794902eb93fe666d5c5", GitTreeState:"clean", BuildDate:"2018-04-19T19:25:45Z", GoVersion:"go1.10.1", Compiler:"gc", Platform:"linux/amd64"}
```

### PR DESCRIPTION
**Pull request by karataliu**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #63226

**Special notes for your reviewer**:

cc @adnavare 

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/cloud/node_controller.go [hunk 1]
```diff
@@ -438,21 +438,24 @@ func excludeTaintFromList(taints []v1.Taint, toExclude v1.Taint) []v1.Taint {
 	return newTaints
 }
 
-// ensureNodeExistsByProviderIDOrInstanceID first checks if the instance exists by the provider id and then by calling instance id with node name
+// ensureNodeExistsByProviderIDOrInstanceID checks if the instance exists by the provider id,
+// If provider id is empty it calls instanceId with node name to get provider id
 func ensureNodeExistsByProviderIDOrInstanceID(instances cloudprovider.Instances, node *v1.Node) (bool, error) {
```
Comment (feiskyer) [inline pkg/controller/cloud/node_controller.go]: Also rename the function to `ensureNodeExistsByProviderID()`?

Comment (karataliu) [reply pkg/controller/cloud/node_controller.go]: Yeah, the name is indeed confusing now. It still indicates the ensure operation involves both providerID and instanceID.

Do we suppose 'instanceID' and 'providerID' to be the same thing? If it is I think it's fine to just name 'ByProviderID'

Currently it seems the case is 'InstanceID' returned by cloud provider is used as 'ProviderID'.
Related:
https://github.com/kubernetes/kubernetes/issues/60466

File: pkg/controller/cloud/node_controller_test.go [hunk 2]
```diff
@@ -121,28 +137,29 @@ func TestEnsureNodeExistsByProviderIDOrNodeName(t *testing.T) {
...
+				fc.ExtID = map[types.NodeName]string{
+					types.NodeName(tc.node.Name): "provider-id://a",
+				}
+			}
+
 			instances, _ := fc.Instances()
-			exists, err := ensureNodeExistsByProviderIDOrInstanceID(instances, tc.node)
-			assert.NoError(t, err)
+			exists, err := ensureNodeExistsByProviderID(instances, tc.node)
+			if tc.providerIDErr == nil {
```
Comment (andrewsykim) [inline pkg/controller/cloud/node_controller_test.go]: Here can we just do
```
assert.Equal(t, err, providerIDErr)
```
??

**Review by feiskyer — COMMENTED** [review]
/lgtm

File: pkg/controller/cloud/node_controller_test.go (hunk 2 above)
Comment (karataliu) [reply pkg/controller/cloud/node_controller_test.go]: updated

**Pull request merged**